src/agent/Makefile

```diff
@@ -41,6 +41,14 @@ ifeq ($(AGENT_POLICY),yes)
     override EXTRA_RUSTFEATURES += agent-policy
 endif
 
+GUEST_PULL ?= no
```

Comment:
@stevenhorsman checking that I understood. The "guest pull" feature has exclusively peer pods as the user, thus it will be turned off by default. On peer pods side, when it is built the kata-agent for inclusion on the podvm, it should flip `GUEST_PULL`  to `yes`.

There is the nydus-snapshotter image pull method to come next which will be supported on peer pods too.  That method doesn't rely on guest-pull though, but both can coexist in peace & love. If a peer pods' user deploys a Kata pod with `io.containerd.cri.runtime-handler: kata-remote` annotation then the nydus-snapshotter that over, otherwise the guest pull is the default behavior. With both nydus-snapshotter and guest-pull, peer pods relies on a single kata-agent built with `GUEST_PULL=yes`.

Reply:
After sending ^ comment I had the amazing idea of look at the discussions on the issue tracking this PR (https://github.com/kata-containers/kata-containers/issues/8103) and @stevenhorsman's question caught me:

 ```How do we decided whether to build the kata-agent with the guest_pull feature, should it be default? If not, how we choose how to package these new changes in kata-deploy? What about the crictl testing?```

At glance I think guest_pull should be disabled by default to avoid kata depending on CoCo. The only user of that feature is peer pods, which doesn't rely on kata-agent built and bundled in kata-deploy; so from this angle kata-deploy shouldn't be changed to handle the new binary. However, if we don't rebuild kata-agent and change kata-deploy then Kata's CI won't be able to proper test that feature.

IMHO the way we ship rebuilds of same components doesn't scale well because it is required to go all way down from the creation of a new runtimeclass.

Reply:
@wainersm Thank you for your review. I’m still learning about peer pods and from what I understand, peer pods requires the kata-agent to be built in a certain way. Could you please explain to me why peer pods doesn’t depend on the kata-agent build?

If the goal is to build the kata-agent only for peer pods without using kata-deploy, I think we can run the command `sudo -E make GUEST_PULL=yes` in the kata-containers/src/agent directory. I hope this helps.

Reply:
Same comment here about the `GUEST_PULL` name.